### PR TITLE
Add an IO interface to readqps

### PR DESCRIPTION
**Pull request by mlubin**
This allows reading from in-memory buffers or gzipped files.

### CODE REVIEW

**Comment by dpo** [comment]
Thanks. That seems good to me and should allow reading compressed files directly.

File: src/readqps.jl [hunk 1]
```diff
@@ -700,6 +700,11 @@ function read_quadobj_line!(qps::QPSData, card::MPSCard)
 end
 
 function readqps(filename::String; mpsformat::Symbol=:free)
+    qps = open(filename, "r")
```
Comment (mtanneau) [inline src/readqps.jl:703]: Shouldn't there be a `close` at some point?

**Comment by mtanneau** [comment]
Thanks Miles :)

File: src/readqps.jl (hunk 1 above)
Comment (mlubin) [reply src/readqps.jl]: Good point. The IO version of `readqps` was closing the stream, but that doesn't seem right. Rearranged this as an open(...) do block.

**Pull request merged**